stdlib/3/functools.pyi

```diff
@@ -78,6 +78,13 @@ class _SingleDispatchCallable(Generic[_T]):
 def singledispatch(func: Callable[..., _T]) -> _SingleDispatchCallable[_T]: ...
 
 if sys.version_info >= (3, 8):
+    class singledispatchmethod:
+        def __init__(self, func: Callable[..., _T]) -> None: ...
```

Comment:
This `_T` doesn't make sense because there is nothing to bind the TypeVar. I'm not sure how this thing should be typed but it should probably involve something similar to `_SingleDispatchCallable` above.

Reply:
Oh I meant to make the class Generic. Is there a reason mypy doesn't complain about this?

It's typed to closely match the implementation, which descriptor protocol aside is pretty straightforward (just added the instance variables in too).